google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkService.java

```diff
@@ -39,6 +40,8 @@ public static CloudSdkService getInstance() {
 
   public abstract void setSdkHomePath(String path);
 
+  public abstract Set<CloudSdkValidationResult> validateCloudSdk(@NotNull Path pathToCloudSdk);
```

Comment:
Checking the version is common to every plugin that uses the library, since they all support the same operations. I'm pretty sure we could completely offload this to the library, but wouldn't recommend doing so in this PR since it can still be quite a workload.